src/BayesFilters/src/utils.cpp

```diff
@@ -16,3 +16,106 @@ double bfl::utils::log_sum_exp(const Ref<const VectorXd>& arguments)
 
     return max + log((arguments.array() - max).exp().sum());
 }
+
+
+MatrixXd bfl::utils::quaternion_to_rotation_vector(const Ref<const MatrixXd>& quaternion)
+{
+    MatrixXd rotation_vectors(3, quaternion.cols());
+    for (std::size_t i = 0; i < quaternion.size(); i++)
+    {
+        double norm_n = quaternion.col(i).tail<3>().norm();
+        if (norm_n > std::numeric_limits<double>::epsilon())
+        {
+            const double& w = quaternion.col(i)(0);
+            if (w < 0)
+                rotation_vectors.col(i) = - 2.0 * std::acos(-w) * quaternion.col(i).tail<3>() / norm_n;
+            else
+                rotation_vectors.col(i) = 2.0 * std::acos(w) * quaternion.col(i).tail<3>() / norm_n;
+        }
+        else
+            rotation_vectors.col(i) = Vector3d::Zero();
+    }
+
+    return rotation_vectors;
+}
+
+
+MatrixXd bfl::utils::rotation_vector_to_quaternion(const Ref<const MatrixXd>& rotation_vector)
+{
+    MatrixXd quaternions(4, rotation_vector.cols());
+    for (std::size_t i = 0; i < rotation_vector.cols(); i++)
+    {
+        double norm_r = rotation_vector.col(i).norm();
+        if (norm_r > std::numeric_limits<double>::epsilon())
```

Comment:
Not sure that `std::numeric_limits<double>::epsilon()` (have a look at its meaning [here](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon)) is the right choice here.
What do you exactly want check here? Maybe `std::numeric_limits<double>::min()` is better?

Reply:
The idea is to detect when I can safely normalize 
`rotation_vector.col(i)`. Using the `Eigen` `normalize()` method is not that good as it returns the vector unchanged if the normalization is not possible, thus I prefer to check it directly. 

In the latest commit, I switched from `norm > std::numeric_limits<double>::epsilon()` to `norm > 0` as it should be safe to normalize the vector as long as a division by a positive number occurs. B.t.w the `Eigen` `normalize()` method does not normalize when `norm() == 0`.